src/backend/opencl/blas.cpp

```diff
@@ -24,10 +24,7 @@
 
 // Includes one of the supported OpenCL BLAS back-ends (e.g. clBLAS, CLBlast)
 #include <magma/magma_blas.h>
-
-#if defined(WITH_LINEAR_ALGEBRA)
 #include <cpu/cpu_blas.hpp>
-#endif
```

Comment:
Shouldn't `gemm_fallback` definition and specilization be placed within these guards instead of removing this guard here. We would be compiling with those headers even when we won't use build with linear algebra.